BraintreeDataCollector/src/test/java/com/braintreepayments/api/DataCollectorUnitTest.java

```diff
@@ -76,4 +77,22 @@ public void collectDeviceData_getsDeviceDataJSONWithCorrelationIdFromPayPal() th
         JSONObject json = new JSONObject(deviceData);
         assertEquals("sample_correlation_id", json.getString("correlation_id"));
     }
+
+    @Test
+    public void collectDeviceData_without_DataCollectorRequest_sets_hasUserLocationConsent_to_false() {
```

Comment:
I think we should have a test for with DataCollectorRequest passes the value from the request (to test the true case)

Reply:
Added the missing test case